docs/pages/ESSupport.md

```diff
@@ -0,0 +1,350 @@
+
+### sap.ui.define/sap.ui.require: Usage of Object Expressions, Spread Elements and Template Literals
+An `Object Expression`, `Spread Element` or  `Template Literal` with an expression is not supported as a parameter for `sap.ui.define` or `sap.ui.require`.
+
+**✅ Best Practice:**
+```javascript
+sap.ui.define(["Bar"], function(Bar){
+});
+```
+
+**⛔️ Not supported: Object Expression**
+```javascript
+const dependency = "Bar";
+sap.ui.define([dependency], function(Bar){
+});
+```
+
+**⛔️ Not supported: Spread Element**
+```javascript
+const dependenciesA = ["ModuleA1", "ModuleA2"];
+const dependenciesB = ["ModuleB1", "ModuleB2"];
+sap.ui.define([...dependenciesA, ...dependenciesB],
+    function(ModuleA1, ModuleA2, ModuleB1, ModuleB2){
+});
+```
+
+**⛔️ Not supported: Template Literal with Expression**
+```javascript
+sap.ui.define([`ModuleA`, `ModuleB` ], () => {
+		const i = `D`;
+		return sap.ui.require([
+			`ModuleC`,
+			`Module${i}`
+		], () => { });
+});
+```
+
+### XMLComposites
+When declaring a `XMLComposite` any `Object Expression`, `Spread Element` or `Template Literal` with expression is not supported.
+
+**✅ Best Practice:**
+```javascript
+sap.ui.define(["sap/ui/core/XMLComposite"],
+	(XMLComposite) => XMLComposite.extend("composites.MyComposite", {
+    });
+```
+
+**⛔️ Not supported: Object Expression**
+```javascript
+sap.ui.define([
+	"sap/ui/core/XMLComposite"],
+		(XMLComposite) => {
+		const key = "fragment"
+		return XMLComposite.extend("composites.MyComposite", {
+			[key]: "composites.custom.MyComposite"
+		});
+	});
+```
+
+**⛔️ Not supported: Spread Element**
+```javascript
+sap.ui.define([
+	"sap/ui/core/XMLComposite"],
+		(XMLComposite) => {
+		const myXMLComposite = {
+			fragment: "composites.custom.MyComposite"
+		};
+		return XMLComposite.extend("composites.MyComposite", {
+			...myXMLComposite
+		});
+	});
+```
+
+**⛔️ Not supported: Template Literal with Expression**
+```javascript
+sap.ui.define(["sap/ui/core/XMLComposite"],
+	(XMLComposite) => {
+        const name = "MyComposite";
+	    return XMLComposite.extend(`composites.${name}`, {});
+```
+
+### Smart Templates
+When declaring a `SmartTemplate` any `Object Expression`, `Spread Element` or `Template Literal` with expression is not supported.
+
+**✅ Best Practice:**
+```javascript
+sap.ui.define(["sap/suite/ui/generic/template/lib/TemplateAssembler"],
+	(TemplateAssembler) => TemplateAssembler.getTemplateComponent(getMethods,
+		"sap.fe.templates.Page.Component", {
+			metadata: {
+				properties: {
+					"templateName": {
+						"type": "string",
+						"defaultValue": "sap.fe.templates.Page.view.Page"
+					}
+				},
+				"manifest": "json"
+			}
+        }
+));
+```
+
+**⛔️ Not supported: Object Expression**
+```javascript
+sap.ui.define(["sap/suite/ui/generic/template/lib/TemplateAssembler"],
+	(TemplateAssembler) => {
+        const key =  "templateName";
+        return TemplateAssembler.getTemplateComponent(getMethods,
```

Comment:
use https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#highlighting-specific-lines for hl desired LOCs

Reply:
done